code/object/objcollide.cpp

```diff
@@ -889,7 +889,7 @@ void obj_collide_pair(object *A, object *B)
         collision_info->next_check_time = timestamp(0);
     }
 
-    if ( valid &&  A->type != OBJ_BEAM ) {
```

Comment:
What's going on here that it's a good idea to remove this check?

Reply:
(my last paragraph in the description)

Reply:
Were you waiting for me to post this comment?

Reply:
Pure happenstance I assure you.

Reply:
Ok.  The reason for the confusion is that I don't see what you're talking about in the third paragraph in or around this line.

Reply:
This is gating the early out for general collision checking if the `next_check_time` is -1. This is done to indicate that the pair will never collide and collision can move on to the next pair immediately. However, even if a previous collision returned 1, to indicate `next_check_time` should be set to -1 and never checked again, if a beam is involved (beams are always `A`) its ignores this early out and attempts collision anyway. As mentioned I see no good reason for this to be the case. It's probably not much of a time cost, but still bizarre and as far as I can tell, unnecessary.

Reply:
Can't remember what happened after your last reply.  Might have fallen asleep or had to take care of my kid.